whois-update/src/main/java/net/ripe/db/whois/update/handler/transform/AsSetSourceAttributeTransformer.java

```diff
@@ -0,0 +1,69 @@
+package net.ripe.db.whois.update.handler.transform;
+
+import net.ripe.db.whois.common.dao.RpslObjectDao;
+import net.ripe.db.whois.common.domain.CIString;
+import net.ripe.db.whois.common.rpsl.AttributeType;
+import net.ripe.db.whois.common.rpsl.ObjectType;
+import net.ripe.db.whois.common.rpsl.RpslAttribute;
+import net.ripe.db.whois.common.rpsl.RpslObject;
+import net.ripe.db.whois.common.rpsl.RpslObjectBuilder;
+import net.ripe.db.whois.update.domain.Action;
+import net.ripe.db.whois.update.domain.Update;
+import net.ripe.db.whois.update.domain.UpdateContext;
+import net.ripe.db.whois.update.domain.UpdateMessages;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+@Component
+public class AsSetSourceAttributeTransformer implements Transformer{
+
+    private final RpslObjectDao rpslObjectDao;
+
+    @Value("${whois.nonauth.source}")
+    private String nonAuthSource;
+
+    public AsSetSourceAttributeTransformer(final RpslObjectDao rpslObjectDao){
+        this.rpslObjectDao = rpslObjectDao;
+    }
+
+    @Override
+    public RpslObject transform(final RpslObject rpslObject, final Update update,
+                                final UpdateContext updateContext,
+                                final Action action) {
+        if (Action.DELETE.equals(action) || !ObjectType.AS_SET.equals(rpslObject.getType())) {
+            return rpslObject;
+        }
+        final String asSetKey = rpslObject.getKey().toString();
+
+        boolean flatAsSet = !asSetKey.contains(":");
+        final CIString asSetSource = rpslObject.getValueForAttribute(AttributeType.SOURCE);
+
+        if (flatAsSet){
+            updateContext.addMessage(update, UpdateMessages.flatModelNotAllowSourceModifications(asSetSource.toString()));
```

Comment:
This message is added for *all* create or update to a flat AS-SET, regardless of source?

Reply:
I am reviewing this, I will try to add a warning and change to the original source if the user try to update the source. I don't have the original object so I don't know the original message source...and I don't like to pass another parameter to the transformer

Reply:
now this message will be triggered when the user is updating an flat as-set and the source is different. We will notify the user that the update request has changed to match the database source